src/network/src/ros/src/broker/CommunicationBroker.cpp

```diff
@@ -0,0 +1,49 @@
+#include "CommunicationBroker.h"
+
+CommunicationBroker::CommunicationBroker() {
+    ROS_INFO("Communication Broker initialization");
+    for (std::uint16_t& robotID : getRobotList()) {
+        std::string subscribingTopic = "/Communication/broadcastOutput/" + std::to_string(robotID);
+        std::string publishingTopic = "/Communication/broadcastInput/" + std::to_string(robotID);
+
+        ros::Publisher pub = m_nodeHandle.advertise<std_msgs::String>(publishingTopic, 1000);
+        ros::Subscriber sub = m_nodeHandle.subscribe(
+            subscribingTopic, 1000, &CommunicationBroker::communicationCallback, this);
+
+        m_publishersMap.emplace(robotID, pub);
+        m_subscribersList.push_back(sub);
+    }
+}
+
+std::vector<uint16_t> CommunicationBroker::getRobotList() {
+    std::vector<uint16_t> robotList;
+    ros::NodeHandle nh("~");
+
+    XmlRpc::XmlRpcValue configRobotList;
+    std::string robotListParamName =
+        nh.param(std::string("configList"), std::string("/Broker/robots"));
+    if (!nh.getParam("/Broker/robots", configRobotList)) {
```

Comment:
What we could do instead of hardcoding is at boot, the broadcast or unicast sends a ros message on a topic with their ID, and the communication broker catches them and stores thoses id. though that could go in another PR, the file is fine for now.

Reply:
I like this idea, we could also use a similar mechanism if we spawn and close dynamically for some reason